index.js

```diff
@@ -1,25 +1,53 @@
 'use strict';
 
 module.exports = function buildAirbnbPreset(context, options) {
+  // This is similar to how `env` works in Babel:
+  // https://babeljs.io/docs/usage/babelrc/#env-option
+  // We are not using `env` because it’s ignored in versions > babel-core@6.10.4:
+  // https://github.com/babel/babel/issues/4539
+  // https://github.com/facebookincubator/create-react-app/issues/720
+  // It’s also nice that we can enforce `NODE_ENV` being specified.
+  var env = process.env.BABEL_ENV || process.env.NODE_ENV;
+  if (env !== 'development' && env !== 'test' && env !== 'production') {
+    throw new Error(
+      'Using `babel-preset-airbnb` requires that you specify `NODE_ENV` or '+
+      '`BABEL_ENV` environment variables. Valid values are "development", ' +
+      '"test", and "production". Instead, received: ' + JSON.stringify(env) + '.'
+    );
+  }
+
+  var allPlugins = [
+    [require('babel-plugin-transform-es2015-template-literals'), { spec: true }],
+    require('babel-plugin-transform-es3-member-expression-literals'),
+    require('babel-plugin-transform-es3-property-literals'),
+    require('babel-plugin-transform-jscript'),
+    require('babel-plugin-transform-exponentiation-operator'),
+    require('babel-plugin-syntax-trailing-function-commas'),
+    [require('babel-plugin-transform-object-rest-spread'), { useBuiltIns: true }],
+    [require('babel-plugin-transform-react-jsx'), { useBuiltIns: true }],
+  ];
   var preset; // eslint-disable-line no-var
+
   if (options && options.modules === false) {
     preset = require('babel-preset-es2015').buildPreset(null, { modules: false });
   } else {
     preset = require('babel-preset-es2015-without-strict');
```

Comment:
unrelated: While you are in here, would you also be interested in inlining these es2015 presets? It would be nice to turn this option on for some perf improvements: https://github.com/facebook/react/pull/9002/files#diff-e56633f72ecc521128b3db6586074d2cR21